src/cli/templates/_worker.js/index.ts

```diff
@@ -0,0 +1,65 @@
+import { handleRequest } from './handleRequest';
+import { adjustRequestForVercel } from './utils';
+import type { AsyncLocalStorage } from 'node:async_hooks';
+
+declare const __NODE_ENV__: string;
+
+declare const __CONFIG__: ProcessedVercelConfig;
+
+declare const __BUILD_OUTPUT__: VercelBuildOutput;
+
+declare const __ENV_ALS_PROMISE__: Promise<null | AsyncLocalStorage<unknown>>;
+
+export default {
+	async fetch(request, env, ctx) {
+		const envAsyncLocalStorage = await __ENV_ALS_PROMISE__;
+		if (!envAsyncLocalStorage) {
+			return new Response(
+				`Error: Could not access built-in Node.js modules. Please make sure that your Cloudflare Pages project has the 'nodejs_compat' compatibility flag set.`,
+				{ status: 503 }
+			);
+		}
+
+		const cloudflareGlobalContextSymbol = Symbol.for(
+			'cloudflare-global-context'
+		);
+
+		const cloudflareGlobalContextAls = (
+			globalThis as unknown as {
+				[cloudflareGlobalContextSymbol]?: AsyncLocalStorage<CloudflareGlobalContext>;
+			}
+		)[cloudflareGlobalContextSymbol];
```

Comment:
unfortunately here we duplicate some code from `src/utils/utils/cloudflareGlobalContext.ts` but I couldn't share it because we don't include `src/utils` in the package (while we do include the `templates` code, which is used by the CLI on build)

I could do something like creating a new shared dir with this code and include it in the package, or move this code to the templates dir and import it from the utils... none of these options sounds very good to me, so I just went with having some code duplication

If anyone has a better idea (or would prefer some of the solutions above) please let me know 🙏